eclair-core/src/main/scala/fr/acinq/eclair/io/OpenChannelInterceptor.scala

```diff
@@ -0,0 +1,75 @@
+import fr.acinq.eclair.channel.{ChannelConfig, LocalParams, SupportedChannelType}
+import fr.acinq.eclair.io.Peer.{ConnectedData, OutgoingMessage, SpawnChannelNonInitiator}
+import fr.acinq.eclair.wire.protocol.{Error, OpenChannel, OpenDualFundedChannel}
+import fr.acinq.eclair.{AcceptOpenChannel, InterceptOpenChannelPlugin, InterceptOpenChannelReceived, InterceptOpenChannelResponse, RejectOpenChannel}
+
+import scala.concurrent.duration.FiniteDuration
+
+// Child actor of a Peer that handles the interactions with an InterceptOpenChannelPlugin:
+//  - sends OpenChannelReceived to the plugin
+//  - waits for its response and translates it to the Peer data format (either SpawnChannelNonInitiator or an outgoing Error)
+//  - reject the channel open after a timeout (configurable in eclair.conf)
+//
+//  Note: we don't fully trust plugins to be correctly implemented, and we need to respond to our peer even if the plugin fails to tell us what to do
+
+object OpenChannelInterceptor {
+  def apply(replyTo: ActorRef[Any], plugin: InterceptOpenChannelPlugin, timeout: FiniteDuration, connectedData: ConnectedData, temporaryChannelId: ByteVector32, localParams: LocalParams, fundingAmount: Satoshi, open: Either[OpenChannel, OpenDualFundedChannel], channelType: SupportedChannelType): Behavior[Command] = {
+    Behaviors.setup { context =>
+      Behaviors.withTimers { timers =>
+        timers.startSingleTimer(PluginTimeout, timeout)
+        new OpenChannelInterceptor(replyTo, plugin, connectedData, temporaryChannelId, localParams, fundingAmount, open, channelType, context).start()
+      }
+    }
+  }
+
+  // @formatter:off
+  sealed trait Command
+  case object PluginTimeout extends Command
+  case class WrappedOpenChannelResponse(pluginResponse: InterceptOpenChannelResponse) extends Command
+  // @formatter:on
+}
+
+private class OpenChannelInterceptor(replyTo: ActorRef[Any], plugin: InterceptOpenChannelPlugin, connectedData: ConnectedData, temporaryChannelId: ByteVector32, localParams: LocalParams, fundingAmount: Satoshi, open: Either[OpenChannel, OpenDualFundedChannel], channelType: SupportedChannelType, context: ActorContext[OpenChannelInterceptor.Command]) {
+
+  import OpenChannelInterceptor._
+
+  private def start(): Behavior[Command] = {
+    val pluginResponseAdapter = context.messageAdapter[InterceptOpenChannelResponse](WrappedOpenChannelResponse)
+    plugin.getOpenChannelInterceptor ! InterceptOpenChannelReceived(pluginResponseAdapter, open, temporaryChannelId, localParams, fundingAmount)
+
+    Behaviors.receiveMessage {
+      case PluginTimeout =>
+        context.log.error(s"plugin ${plugin.name} timed out while intercepting open channel")
+        replyTo ! OutgoingMessage(Error(temporaryChannelId, "plugin timeout"), connectedData.peerConnection)
+        Behaviors.stopped
+      case WrappedOpenChannelResponse(a: AcceptOpenChannel) =>
+        replyTo ! SpawnChannelNonInitiator(open, ChannelConfig.standard, channelType, a.localParams)
```

Comment:
The channel config should probably be provided to this actor, like we do for the channel type, otherwise if we introduce a different one than `standard` later we'll probably forget to update this piece of code.

Reply:
Fixed in [9c72ae8](https://github.com/ACINQ/eclair/pull/2552/commits/9c72ae850dee42e0e3ba5db1f205a8c8ceda46a9)